genienlp/models/transformer_token_classification.py

```diff
@@ -0,0 +1,90 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+logging.getLogger("filelock").setLevel(logging.WARNING)
+
+from transformers import AutoModelForTokenClassification, AutoConfig
+
+from ..models.base import GenieModel
+from ..util import adjust_language_code
+from . import TransformerSeq2Seq
+from ..data_utils.numericalizer import TransformerNumericalizer
+
+logger = logging.getLogger(__name__)
+
+
+class TransformerForTokenClassification(TransformerSeq2Seq, GenieModel):
+    def __init__(self, config=None, *inputs, args, tasks, vocab_sets, save_directory=None, **kwargs):
+    
+        num_labels = 0
+        if args.num_labels is not None:
+            num_labels = args.num_labels
+        else:
+            for task in tasks:
+                # if having multiple tasks choose max num_labels
+                if hasattr(task, 'num_labels'):
+                    num_labels = max(num_labels, task.num_labels)
+        
+        config = AutoConfig.from_pretrained(args.pretrained_model, cache_dir=args.embeddings, num_labels=num_labels, finetuning_task='ned')
+        GenieModel.__init__(self, config)
+        self.args = args
+        if hasattr(config, 'd_model'):
+            args.dimension = config.d_model
+        else:
+            args.dimension = config.hidden_size
+
+        self.src_lang, self.tgt_lang = adjust_language_code(config, args.pretrained_model,
+                                                            kwargs.get('src_lang', 'en'), kwargs.get('tgt_lang', 'en'))
+
+        if save_directory is not None:
+            self.model = AutoModelForTokenClassification.from_config(config)
+        else:
+            self.model = AutoModelForTokenClassification.from_pretrained(self.args.pretrained_model,
+                                                                         cache_dir=self.args.embeddings,
+                                                                         config=config)
+
+        self.numericalizer = TransformerNumericalizer(self.args.pretrained_model, args, max_generative_vocab=None,
+                                save_dir=save_directory, config=config, src_lang=self.src_lang, tgt_lang=self.tgt_lang, vocab_sets=vocab_sets, tasks=tasks)
+        
+        self.model.resize_token_embeddings(self.numericalizer.num_tokens)
+        
+        self.numericalizer.answer_pad_id = -100
```

Comment:
Is this value the same for all models? And shouldn't this code be moved to the numericalizer class?

Reply:
`answer_pad_id` defaults to `pad_id` for all models and is set in `numericalizer.py` line 393. I overwrite it only for TransformerForTokenClassification models.

Reply:
Why is it necessary to be overridden?

Reply:
In token classification, model generates a number in the range [0, num_labels) so it's convenient to have a different pad_id for answers other than the conventional pad_id=0